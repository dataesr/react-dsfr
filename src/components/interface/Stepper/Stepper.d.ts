import * as React from 'react';

export type StepperClassName = string | Object | any[];

export interface StepperProps {
    steps: number;
    currentStep: number;
    className?: StepperClassName;
}

declare const Stepper: React.FC<StepperProps>;

export default Stepper;

