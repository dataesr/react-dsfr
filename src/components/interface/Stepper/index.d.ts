import type Stepper from './Stepper';

export type { Stepper };
