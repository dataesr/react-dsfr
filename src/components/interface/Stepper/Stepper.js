import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import dataAttributes from '../../../utils/data-attributes';

import '@gouvfr/dsfr/dist/component/stepper/stepper.css';

const Stepper = ({
  currentStep, steps, currentTitle, nextStepTitle, className, ...remainingProps
}) => {
  const _className = classNames('fr-stepper', className, {});

  return (
    <div className={_className} {...dataAttributes.getAll(remainingProps)}>
      <h2 className="fr-stepper__title">
        <span className="fr-stepper__state">{`Étape ${currentStep} sur ${steps}`}</span>
        {currentTitle}
      </h2>
      <div className="fr-stepper__steps" data-fr-current-step={currentStep} data-fr-steps={steps} />
      <p className="fr-stepper__details">
        <span className="fr-text--bold">Étape suivante : </span>
        {nextStepTitle}
      </p>
    </div>
  );
};

Stepper.propTypes = {
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array,
  ]),
  currentStep: PropTypes.number.isRequired,
  steps: PropTypes.number.isRequired,
  currentTitle: PropTypes.string.isRequired,
  nextStepTitle: PropTypes.string.isRequired,
};

Stepper.defaultProps = {
  className: '',
};

export default Stepper;
