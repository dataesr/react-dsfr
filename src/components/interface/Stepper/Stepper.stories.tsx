import { ComponentStory } from '@storybook/react';

import Stepper from './Stepper';

export default {
  title: 'Composants/Stepper',
  component: Stepper,
  argTypes: {},
  parameters: {
    docs: {
      description: {
        component: "L'indicateur d'étapes permet à l'utilisateur de se repérer dans un formulaire.",
      },
    },
  },
};

const Template: ComponentStory<typeof Stepper> = (args) => <Stepper {...args} />;

export const FirstStep = Template.bind({});
FirstStep.args = {
  steps: 4,
  currentStep: 1,
  currentTitle: 'Titre de l\'étape en cours',
  nextStepTitle: 'Titre de la prochaine étape',
};
FirstStep.storyName = "Indicateur d'étapes";

export const LastStep = Template.bind({});
LastStep.args = {
  steps: 4,
  currentStep: 4,
  currentTitle: 'Titre de l\'étape en cours',
  nextStepTitle: 'Titre de la prochaine étape',
};
LastStep.storyName = "Indicateur d'étapes - Dernière étape";
