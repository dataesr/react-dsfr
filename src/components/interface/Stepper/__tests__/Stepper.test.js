import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import React from 'react';
import Stepper from '../Stepper';

Enzyme.configure({ adapter: new Adapter() });

const initialProps = {
  currentStepTitle: 'TiTle',
  nextStepTitle: 'TiTle next',
  steps: 6,
  currentStep: 5,
};

describe('<Stepper />', () => {
  it('renders correctly', () => {
    const component = renderer
      .create(
        <Stepper
          currentStep={initialProps.currentStep}
          steps={initialProps.steps}
          currentTitle={initialProps.currentStepTitle}
          nextStepTitle={initialProps.nextStepTitle}
        />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
