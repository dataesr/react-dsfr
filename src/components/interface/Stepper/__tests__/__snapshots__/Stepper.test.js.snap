// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stepper /> renders correctly 1`] = `
<div
  className="fr-stepper"
>
  <h2
    className="fr-stepper__title"
  >
    <span
      className="fr-stepper__state"
    >
      Étape 5 sur 6
    </span>
    TiTle
  </h2>
  <div
    className="fr-stepper__steps"
    data-fr-current-step={5}
    data-fr-steps={6}
  />
  <p
    className="fr-stepper__details"
  >
    <span
      className="fr-text--bold"
    >
      Étape suivante : 
    </span>
    TiTle next
  </p>
</div>
`;
